crypto-ffi/src/generic/mod.rs

```diff
@@ -954,20 +954,38 @@ struct CoreCryptoLoggerWrapper {
     logger: std::sync::Arc<dyn CoreCryptoLogger>,
 }
 
+impl CoreCryptoLoggerWrapper {
+    fn adjusted_log_level(&self, metadata: &Metadata) -> Level {
+        match (metadata.level(), metadata.target()) {
+            // increase log level for refinery_core::traits since they are too verbose in transactions
+            (level, "refinery_core::traits") if level >= Level::Info => Level::Debug,
+            (level, "refinery_core::traits::sync") if level >= Level::Info => Level::Debug,
+            (level, _) => level,
+        }
+    }
+}
+
 impl log::Log for CoreCryptoLoggerWrapper {
-    fn enabled(&self, _metadata: &Metadata) -> bool {
-        true
+    fn enabled(&self, metadata: &Metadata) -> bool {
+        log::max_level() >= self.adjusted_log_level(metadata)
     }
 
     fn log(&self, record: &Record) {
         let kvs = record.key_values();
         let mut visitor = KeyValueVisitor(BTreeMap::new());
         let _ = kvs.visit(&mut visitor);
 
+        if !self.enabled(record.metadata()) {
+            return;
+        }
```

Comment:
Is this really necessary? I would have expected that the call to `enabled()` is done internally in the logger?

Reply:
One would think so but the docs say:
> This method isn’t called automatically by the log! macros. It’s up to an implementation of the Log trait to call enabled in its own log method implementation to guarantee that filtering is applied.